libs/filer/filer.go

```diff
@@ -7,13 +7,22 @@ import (
 	"io/fs"
 )
 
+// WriteMode captures intent when writing a file.
+//
+// The first 9 bits are reserved for the [fs.FileMode] permission bits.
+// These are used only by the local filer implementation and have
+// no effect for the other implementations.
 type WriteMode int
 
+// writeModePerm is a mask to extract permission bits from a WriteMode.
+const writeModePerm = WriteMode(fs.ModePerm)
+
 const (
-	OverwriteIfExists WriteMode = 1 << iota
+	OverwriteIfExists WriteMode = (writeModePerm + 1) << iota
```

Comment:
Can you add a comment on why we start from the 10th bit. That the first 9 are reserved for rwx file permissions.

Reply:
Done.